src/csharp/Grpc.Core/Internal/DefaultDeserializationContext.cs

```diff
@@ -20,47 +20,85 @@
 using System;
 using System.Threading;
 
+#if GRPC_CSHARP_SUPPORT_SYSTEM_MEMORY
+using System.Buffers;
+#endif
+
 namespace Grpc.Core.Internal
 {
     internal class DefaultDeserializationContext : DeserializationContext
```

Comment:
I'm not familiar with DeserializationContext. Some quick questions to make sure I understand it correctly:
1. DefaultDeserializationContext is specific to C Core. Grpc.AspNetCore.Server would create an implementation of the context that reads from a PipeReader. Is the content of the sequence returned by PayloadAsReadOnlySequence the same as the content passed to Marshaller.Deserializer? (no header, decompression - if required - has already been run)
2. DeserializationContext is stateful. It looks like C Core creates a thread local copy. How instances of the context are created and reused is completely up to Grpc.AspNetCore.Server, correct?
3. Codegened services currently always use Marshaller.Create that takes funcs that work with byte[]. To use the new PayloadAsReadOnlySequence we would need the protobuf framework to support reading from the sequence, and codegen tools to be updated to produce code that uses the serialization/deserialization contexts. Correct?

Reply:
1. yes it's the message payload (after being decompressed, if applicable), there's no access to the header.  Usually this is the serialized form of a protobuf message.
2. yes, creating instances of DeserializationContext is opaque for the user. Having a thread local copy and reusing is fine because the deserializationContext is only accessible from marshaller.ContextualDeserializer function.
3. correct, we would need codegen changes to use the contextual serializer / deserializer and invoke the right protobuf parse and serialize methods in order to benefit from this new API.  One detail is that for the current way the PR is implemented, getting the byte[] payload on netstandard2.0 actually uses PayloadAsReadOnlySequence internally.